sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -65,7 +65,7 @@ def raise_with_traceback(exception, *args, **kwargs):
     message = kwargs.pop("message", "")
     exc_type, exc_value, exc_traceback = sys.exc_info()
     # If not called inside a "except", exc_type will be None. Assume it will not happen
-    exc_msg = "{}, {}: {}".format(message, getattr(exc_type, '__name__'), exc_value)
+    exc_msg = "{}, {}: {}".format(message, exc_type.__name__, exc_value) # type: ignore
```

Comment:
Still don't like the type: ignore, if you think exc_type must not be None (which is what the docstring and the comment is suggesting), add
```python
if exc_type is None:
   raise ValueError("raise_with_traceback can only be used in except clauses")
```